python/cudf/cudf/tests/test_groupby.py

```diff
@@ -1561,3 +1561,34 @@ def test_groupby_unique(by, data, dtype):
     expect = pdf.groupby("by")["data"].unique()
     got = gdf.groupby("by")["data"].unique()
     assert_eq(expect, got)
+
+
+@pytest.mark.parametrize("nelem", [2, 3, 100, 1000])
+@pytest.mark.parametrize("func", ["cummin", "cummax", "cumcount", "cumsum"])
+def test_groupby_2keys_scan(nelem, func):
+    pdf = make_frame(pd.DataFrame, nelem=nelem)
+    expect_df = pdf.groupby(["x", "y"], sort=True).agg(func)
+    got_df = (
+        make_frame(DataFrame, nelem=nelem)
+        .groupby(["x", "y"], sort=True)
+        .agg(func)
+    )
+    # pd.groupby.cumcount returns a series.
+    if isinstance(expect_df, pd.Series):
+        expect_df = expect_df.to_frame("val")
```

Comment:
Is this something that we should be matching the behavior of Pandas on?

Reply:
pandas gives `cumcount` for entire groupby object. Not on individual columns.

Reply:
should this be `groupby::count()` instead?

Reply:
There are a few cases to consider here:

1. `df.groupby(...).cumcount()`
2. `df.groupby(...).agg('cumcount')`
3.  `df.groupby(...).agg(['cumcount'])`

Pandas returns a Series for (1) and (2), and a `DataFrame` for (3). We currently return a `DataFrame` in all cases.

(1) is easy for us to do: we define a `cumcount()` method and make sure we return a `Series`. (2) is less trivial for us:  one of the first things we do is normalize the argument passed to `agg()` into a dictionary. Thus, we lose information about whether the user passed a string, a list or a dictionary to `agg()`. 

Handling (2) is also not necessarily relevant to this PR: we have the exact same problem for the `size` aggreegation as well.

My suggestion is let's make sure we handle (1) correctly for now, and not worry about (2) in this PR.

Reply:
Can you raise an issue about (2)?

Agreed lets tackle it in a follow up

Reply:
@shwina did you already write up the issue? If so, this conversation can be resolved.